hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/impl/ContainerData.java

```diff
@@ -675,4 +548,151 @@ public String toString() {
         + ", ri=" + replicaIndex
         + ", origin=[dn_" + originNodeId + ", pipeline_" + originPipelineId + "])";
   }
+
+  /**
+   * Block byte used, block count and pending deletion count.
+   * This class is immutable.
+   */
+  public static class BlockByteAndCounts {
+    private final long bytes;
+    private final long count;
+    private final long pendingDeletion;
+
+    public BlockByteAndCounts(long bytes, long count, long pendingDeletion) {
+      this.bytes = bytes;
+      this.count = count;
+      this.pendingDeletion = pendingDeletion;
+    }
+
+    public long getBytes() {
+      return bytes;
+    }
+
+    public long getCount() {
+      return count;
+    }
+
+    public long getPendingDeletion() {
+      return pendingDeletion;
+    }
+  }
+
+  /**
+   * Read/write/block statistics of a container.
+   * This class is thread-safe -- all methods are synchronized.
+   */
+  public static class Statistics {
+    private long readBytes;
```

Comment:
Instead of having all methods synchronized. Do you think we can use AtomicReference to update linked fields together or have separate locks for different fields. For instance we need not have any snchronization between [writeBytes.writeCount] and [readBytes, readCount]. Maybe we can have small subclass for storing ByteCount and we can have synchronization inside this subclass.

Reply:
@swamirishi , thanks for review this.

These are just a few `long` values.  Using separated locks
- make the code complicated,
- make it easier to have bugs,
- and may not have significant improvement.